File: 2232-adding-spaces-to-a-string/2232-adding-spaces-to-a-string.cs
public class Solution {
    public string AddSpaces(string s, int[] spaces) 
    {
        StringBuilder ans = new StringBuilder();
        for (int index = 0, j = 0; index < s.Length; index++)
        {
            if (j < spaces.Length && index == spaces[j]) {
                ans.Append(' ');
                j++;
            }
            ans.Append(s[index]);
        }
        return ans.ToString();
    }
}

File: 2232-adding-spaces-to-a-string/2232-adding-spaces-to-a-string.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) 
    {
        string ans = "";
        for (int index = 0, j = 0; index < s.size(); index++)
        {
            if (j < spaces.size() && index == spaces[j]) {
                ans += ' ';
                j++;
            }
            ans += s[index];
        }
        return ans;
    }
};